packages/oauth/src/stores/file-store.ts

```diff
@@ -0,0 +1,116 @@
+import { Installation, InstallationStore, InstallationQuery } from '../index';
+import { Logger } from '../logger';
+import fs from 'fs';
+import path from 'path';
+import { homedir } from 'os';
+
+export interface FileInstallationOptions {
+  baseDir?: string;
+  historicalDataEnabled?: boolean;
+  clientId?: string;
+}
+
+export class FileInstallationStore implements InstallationStore {
+
+  private baseDir: string;
+  private historicalDataEnabled: boolean;
+
+  constructor({
+    baseDir = `${homedir()}/.bolt-app-installation`,
+    clientId,
+    historicalDataEnabled = true,
+  }: FileInstallationOptions = {}) {
+    this.baseDir = clientId !== undefined ? `${baseDir}/${clientId}` : baseDir;
+    this.historicalDataEnabled = historicalDataEnabled;
+  }
+
+  public async storeInstallation(installation: Installation, logger?: Logger): Promise<void> {
+    const { enterprise, team, user } = installation;
+    const installationData = JSON.stringify(installation);
+    const installationDir = this.getInstallationDir(enterprise?.id, team?.id);
+
+    if (logger !== undefined) { logger.warn('Storing installation in FileInstallationStore'); }
+
+    // Create Installation Directory
+    fs.mkdir(installationDir, { recursive: true }, (err) => {
+      if (err !== null) { return console.error(err); }
+    });
+
+    try {
+      writeToFile(`${installationDir}/app-latest`, installationData);
+      writeToFile(`${installationDir}/user-${user.id}-latest`, installationData);
+
+      if (this.historicalDataEnabled) {
+        const currentUTC = Date.now();
+        writeToFile(`${installationDir}/app-${currentUTC}`, installationData);
+        writeToFile(`${installationDir}/user-${user.id}-${currentUTC}`, installationData);
```

Comment:
I mentioned this in the summary, but this is where it's applicable, so carrying it over:

> The Python FileStoreInstallation goes a step further by saving the same information into both installer-latest / installer-[currentUTC] and the user-[user_id]-* records. I omitted the former records because it wasn't clear to me how the duplication was value-adding, but am open to introducing it if folks feel it's important or if I've misunderstood its purpose.

Reply:
Specifically talking about this file based implementation, not having all the historical data is also fine as most developers won't use it for production.

The reason why we store all installations in Python / Java is that having the data can be helpful for troubleshooting / system operations in production. When you do investigation about an issue related to a user who installed the app multiple times, checking each installation's timestamp, the difference on its scopes, and the existence of enterprise_id may help.

If we think this file based one is a kind of reference implementation, having this consideration may be worth considering. Not a strong opinion here, though. Just having a comment about this context there is also fine.

Reply:
Totally get it and am with you regarding the importance. 👍🏼  

I think what I'm not understanding is the difference between [these two records](https://github.com/slackapi/python-slack-sdk/blob/2a4487c81fa95d1cb07a3d916887ac10d67d79ab/slack_sdk/oauth/installation_store/file/__init__.py#L52-L63), other than the extra formatting done to the first, and why having both is necessary or beneficial.

Reply:
Should we consider setting saving historical data default to false? Users could set it to true if they want to save historical data.

Reply:
I don't have a strong opinion on this. Happy to go with whatever others deem the best default.

Reply:
>I think what I'm not understanding is the difference between these two records, other than the extra formatting done to the first, and why having both is necessary or beneficial.

Even after revoking all the user installations, bot token must still exist. This is one of the major improvements in GBP (Granular Bot Permissions). This is the reason why Python / Java SDK have bots table separately. 

If there is a better alternative way, we can go with a different approach in Node SDK first and implement the same in others, though. Having `is_user_token_revoked` flag in the installation table plus trying to fetch a bot token even when all installations are marked as deleted may work. Do we think this is a better approach? 🤔 

>Should we consider setting saving historical data default to false?

In Python / Java SDK, the historical data support is enabled by default. For consistency and encouraging a similar approach for production app design, I prefer turning it on but I am open to change this default for Node SDK (and possibly for others in the future major versions).

Reply:
@misscoded @stevengill For enabling the historical data support by default or not, I'm happy to leave the decision to you all. I've already shared my thoughts on it in the above comment. If we want to change the defaults for Python / Java, we may want to hold off doing so until the next major releases (we won't do at least this year for both).

For having bots and user installations, as I mentioned above, having something similar (or user_token_revocation flag) should be necessary to keep a bot token even after the user(s) who installed the app are deactivated or all the user tokens are revoked. For these two store implementations, I think it's okay not to support the use case.

Reply:
I think it makes sense to have historical data enabled as the default, especially to maintain parity between the three languages. So we're good on that point! I'll keep it as is.

I'd like to revisit the other point you bring up when I circle back around to sort out the lack of support for user tokens as a whole, unless you think we shouldn't move forward with introducing this until that's sorted. I think I'm still fuzzy on the details.